ray-operator/apis/ray/v1alpha1/rayjob_types.go

```diff
@@ -48,7 +48,7 @@ type RayJobSpec struct {
 	// It's only working when ShutdownAfterJobFinishes set to true.
 	TTLSecondsAfterFinished *int32 `json:"ttlSecondsAfterFinished,omitempty"`
 	// RayClusterSpec is the cluster template to run the job
-	RayClusterSpec RayClusterSpec `json:"rayClusterSpec,omitempty"`
```

Comment:
This is an API change and I need to involve @akanso and other major users here @pingsutw @DmitriGekhtman @daikeshi here.

Reply:
A pointer type (nil default) makes sense to me.
RayJobs are new enough that I hope we have the flexibility to make this change.

There's some danger of messing up external integrations which manipulate RayJob structs programmatically.
(Are there other ways in which this change could result in backwards incompatibility?)
@pingsutw any concerns for Flyte integration?

Reply:
\cc @harryge00

Reply:
Since we're in beta, we can still update our plugin if CRD changes.

Reply:
Pointer is ok for me @Basasuya